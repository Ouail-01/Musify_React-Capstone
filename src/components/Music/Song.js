import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import {
  FaPlay, FaPause, FaStar, FaRegStar,
} from 'react-icons/fa';
import '../../styles/Song.css';

export default function Song() {
  const music = useSelector((state) => state.music);
  const audio = new Audio(music.music.preview);
  const [clicked, setClicked] = useState(false);

  const play = () => {
    audio.play();
  };

  const pause = () => {
    audio.pause();
  };

  const saveSong = () => {
    if (clicked) {
      localStorage.removeItem(music.music.title);
      setClicked(false);
    } else {
      localStorage.setItem(music.music.title, JSON.stringify(music.music));
      setClicked(true);
    }
  };

  useEffect(() => {
    play();
    return () => {
      pause();
    };
  }, [audio, music]); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      {music.music.title && (
        <>
          <div className="trackSong">
            <h2>{music.music.title}</h2>
            <div>
              <FaPlay onClick={play} />
              <FaPause onClick={pause} className="pause" />
              {clicked ? (
                <FaStar onClick={saveSong} />
              ) : (
                <FaRegStar onClick={saveSong} />
              )}
            </div>
          </div>
        </>
      )}
    </>
  );
}
