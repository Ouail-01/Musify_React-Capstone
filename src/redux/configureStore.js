import { configureStore } from '@reduxjs/toolkit';
import searchReducer from './search/searchSlice';
import musicReducer from './music/musicSlice';
import artistsReducer from './music/artistesSlice';
import albumReducer from './music/albumSlice';

const store = configureStore({
  reducer: {
    search: searchReducer,
    music: musicReducer,
    artists: artistsReducer,
    albums: albumReducer,
  },
});

export default store;
