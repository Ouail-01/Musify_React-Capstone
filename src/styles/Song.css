.trackSong {
  position: sticky;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  background-color: #3c3c3c;
  bottom: 0;
  height: 70px;
  opacity: 0.9;
}

.trackSong h2 {
  margin-left: 20px;
  text-align: center;
}

.trackSong div {
  margin-right: 20px;
  font-size: 25px;
  display: flex;
}

.trackSong .pause {
  margin-left: 10px;
  margin-right: 10px;
}
