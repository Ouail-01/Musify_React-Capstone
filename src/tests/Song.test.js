import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Song from '../components/Music/Song';

const mockStore = configureStore([]);

describe('Song component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      music: {
        music: {
          preview: 'http://preview-url.com',
          title: 'Song Title',
        },
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <Song />
      </Provider>,
    );
  });

  it('should match the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
