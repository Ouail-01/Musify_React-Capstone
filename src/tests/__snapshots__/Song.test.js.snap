// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song component should match the snapshot 1`] = `
<div
  className="trackSong"
>
  <h2>
    Song Title
  </h2>
  <div>
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
      />
    </svg>
    <svg
      className="pause"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M144 479H48c-26.5 0-48-21.5-48-48V79c0-26.5 21.5-48 48-48h96c26.5 0 48 21.5 48 48v352c0 26.5-21.5 48-48 48zm304-48V79c0-26.5-21.5-48-48-48h-96c-26.5 0-48 21.5-48 48v352c0 26.5 21.5 48 48 48h96c26.5 0 48-21.5 48-48z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      />
    </svg>
  </div>
</div>
`;
